code/modules/integrated_electronics/subtypes/reagents.dm

```diff
@@ -538,3 +538,318 @@
+	var/turf/T2 = get_step(T,turn(direction, -90))
+	var/list/the_targets = list(T,T1,T2)
+	busy = TRUE
+
+	// Create list with particles and their targets
+	var/list/water_particles=list()
+	for(var/a=0, a<5, a++)
+		var/obj/effect/particle_effect/water/W = new /obj/effect/particle_effect/water(get_turf(src))
+		water_particles[W] = pick(the_targets)
+		var/datum/reagents/R = new/datum/reagents(5)
+		W.reagents = R
+		R.my_atom = W
+		reagents.trans_to(W,1)
+
+	//Make em move dat ass, hun
+	addtimer(CALLBACK(src, /obj/item/integrated_circuit/reagent/extinguisher/proc/move_particles, water_particles), 2)
+
+//This whole proc is a loop
+/obj/item/integrated_circuit/reagent/extinguisher/proc/move_particles(var/list/particles, var/repetitions=0)
+	//Check if there's anything in here first
+	if(!particles || particles.len == 0)
+		return
+	// Second loop: Get all the water particles and make them move to their target
+	for(var/obj/effect/particle_effect/water/W in particles)
+		var/turf/my_target = particles[W]
+		if(!W)
+			continue
+		step_towards(W,my_target)
+		if(!W.reagents)
+			continue
+		W.reagents.reaction(get_turf(W))
+		for(var/A in get_turf(W))
+			W.reagents.reaction(A)
+		if(W.loc == my_target)
+			break
+	if(repetitions < 4)
+		repetitions++	//Can't have math operations in addtimer(CALLBACK())
+		addtimer(CALLBACK(src, /obj/item/integrated_circuit/reagent/extinguisher/proc/move_particles, particles, repetitions), 2)
+	else
+		push_data()
+		activate_pin(2)
+		busy = FALSE
+
+
+// - Drain circuit - //
+/obj/item/integrated_circuit/reagent/drain
+	name = "chemical drain circuit"
+	desc = "This circuit either eliminates reagents by creating a puddle or can suck up chemicals on tiles."
+	icon_state = "injector"
+	extended_desc = "Set mode to FALSE to eliminate reagents and TRUE to drain."
+
+	volume = 20
+
+	complexity = 10
+	inputs = list(
+		"mode" = IC_PINTYPE_BOOLEAN
+		)
+	outputs = list(
+		"volume" = IC_PINTYPE_NUMBER,
+		"self reference" = IC_PINTYPE_SELFREF
+		)
+	activators = list(
+		"drain" = IC_PINTYPE_PULSE_IN,
+		"on drained" = IC_PINTYPE_PULSE_OUT,
+		"on fail" = IC_PINTYPE_PULSE_OUT
+		)
+	spawn_flags = IC_SPAWN_DEFAULT|IC_SPAWN_RESEARCH
+	power_draw_per_use = 15
+
+
+/obj/item/integrated_circuit/reagent/drain/Initialize()
+	.=..()
+	set_pin_data(IC_OUTPUT,2, src)
+
+
+/obj/item/integrated_circuit/reagent/drain/do_work()
+	if(get_pin_data(IC_OUTPUT, 1, reagents.total_volume))
+		if(!reagents || !reagents.total_volume)
+			push_data()
+			activate_pin(3)
+			return
+		// Put the reagents on the floortile the assembly is on
+		reagents.reaction(get_turf(src))
+		reagents.clear_reagents()
+		push_data()
+		activate_pin(2)
+		return
+
+	else
+		if(reagents)
+			if(reagents.total_volume >= volume)
+				push_data()
+				activate_pin(3)
+				return
+		// Favorably, drain it from a chemicals pile, else, try something different
+		var/obj/effect/decal/cleanable/drainedchems = locate(/obj/effect/decal/cleanable) in get_turf(src)
```

Comment:
We dont have chempiles here, this has barely any effect since the number of decals that have reagents here are few and far between

Reply:
Gotcha, I'll probably get that dealt with tomorrow (As much as I'd like these circuits asap, I'm the big tired)